python/lsst/ip/isr/measureCrosstalk.py

```diff
@@ -266,13 +225,81 @@ def run(self, exposure, dataId=None):
                       sum(len(jj) for ii in ratios for jj in ii if jj is not None), dataId)
         return ratios
 
+    def extractCrosstalkRatios(self, exposure, threshold=None, badPixels=None):
+        """Extract crosstalk ratios between different amplifiers.
+
+        For pixels above ``threshold``, we calculate the ratio between
+        each background-subtracted target amp and the source amp. We
+        return a list of ratios for each pixel for each target/source
+        combination, as a matrix of lists.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Exposure for which to measure crosstalk.
+        threshold : `float`, optional
+            Lower limit on pixels for which we measure crosstalk.
+        badPixels : `list` of `str`, optional
+            Mask planes indicating a pixel is bad.
+
+        Returns
+        -------
+        ratios : `list` of `list` of `numpy.ndarray`
+           A matrix of pixel arrays. ``ratios[i][j]`` is an array of
+           the fraction of the ``j``-th amp present on the ``i``-th amp.
+           The value is `None` for the diagonal elements.
+
+        Notes
+        -----
+        This has been moved into MeasureCrosstalkTask to allow for easier
+        debugging.
+
+        The lsstDebug.Info() method can be rewritten for __name__ =
+        `lsst.ip.isr.measureCrosstalk`, and supports the parameters:
+
+        debug.display['extract'] : `bool`
+            Display the exposure under consideration, with the pixels used
+            for crosstalk measurement indicated by the DETECTED mask plane.
+        debug.display['pixels'] : `bool`
+            Display a plot of the ratio calculated for each pixel used in this
+            exposure, split by amplifier pairs.  The median value is listed
+            for reference.
+        """
+        if threshold is None:
+            threshold = self.config.threshold
+        if badPixels is None:
+            badPixels = list(self.config.badMask)
+
+        mi = exposure.getMaskedImage()
+        FootprintSet(mi, Threshold(threshold), "DETECTED")
+        detected = mi.getMask().getPlaneBitMask("DETECTED")
+        bad = mi.getMask().getPlaneBitMask(badPixels)
+        bg = calculateBackground(mi, badPixels + ["DETECTED"])
+
+        self.debugView('extract', exposure)
+
+        ccd = exposure.getDetector()
+        ratios = [[None for iAmp in ccd] for jAmp in ccd]
+
+        for ii, iAmp in enumerate(ccd):
+            iImage = mi[iAmp.getBBox()]
+            iMask = iImage.mask.array
+            select = (iMask & detected > 0) & (iMask & bad == 0) & np.isfinite(iImage.image.array)
+            for jj, jAmp in enumerate(ccd):
+                if ii == jj:
+                    continue
+                jImage = extractAmp(mi.image, jAmp, iAmp.getReadoutCorner(), isTrimmed=True)
+                ratios[jj][ii] = (jImage.array[select] - bg)/iImage.image.array[select]
```

Comment:
I haven't really thought about this, but just checking you're sure you don't want to subtract the background from the other image. No need for a lengthy explanation, just wanted to check this was thought about/deliberate.

Reply:
Yes, it's deliberate.